src/controllers/User.js

```diff
@@ -92,69 +92,80 @@ const register = async (req, res) => {
       token,
       expiresIn: process.env.JWT_LIFETIME, // extract the expiration time from the token
     });
+
+    // for testing
+    const expiresInTimestamp = 10800;
+    const expirationDate = new Date(expiresInTimestamp * 1000); // Multiply by 1000 to convert to milliseconds
+    console.log("expirationDate in Register is: ", expirationDate);
+
   } catch (error) {
     console.error(error);
-    res.status(500).json({ message: "Internal server error" });
+    res.status(500).json({ message: 'Internal server error' });
   }
 };
 
 const login = async (req, res) => {
-  const { email, password } = req.body;
-
-  try {
-    const user = await User.findOne({ email });
-
-    if (!user) {
-      return res.status(401).json({ message: "Invalid email or password" });
+    const { email, password } = req.body;
+
+    try {
+      const user = await User.findOne({ email });
+
+      if (!user) {
+          return res.status(401).json({ message: 'Invalid email or password' });
+      }
+
+      const isMatch = await bcrypt.compare(password, user.password);
+
+      if (!isMatch) {
+          return res.status(401).json({ message: 'Invalid email or password' });
+      }
+
+      // Create a JWT - JSON Web Token
+      const token = jwt.sign({ userId: user._id }, process.env.JWT_SECRET, {
+          expiresIn: process.env.JWT_LIFETIME,
+      });
+
+      req.user = { userId: user._id };
+
+      res.setHeader(
+          'Set-Cookie',
+          cookies.serialize('token', token, {
+              httpOnly: true,
+              secure: process.env.NODE_ENV === 'production',
+              sameSite: 'strict',
+              maxAge: 60 * 60 * 8, // 24 hours 
```

Comment:
This should match the token expiration not be hard-coded as 24 hours.

Reply:
updated to match the token expiration as requested